src/optimagic/typing.py

```diff
@@ -12,3 +16,88 @@ class AggregationLevel(Enum):
     SCALAR = "scalar"
     LEAST_SQUARES = "least_squares"
     LIKELIHOOD = "likelihood"
+
+
+@dataclass(frozen=True)
+class DictLikeAccess:
+    """Useful base class for replacing string-based dictionaries with dataclass
+    instances and keeping backward compatability regarding read access to the data
+    structure."""
+
+    def __getitem__(self, key: str) -> Any:
+        if key in self.__dict__:
+            return getattr(self, key)
+        else:
+            raise KeyError(f"{key} not found in {self.__class__.__name__}")
+
+    def __iter__(self) -> Iterator[str]:
+        return iter(self._dict_repr())
+
+    def _dict_repr(self) -> dict[str, Any]:
+        return {field.name: getattr(self, field.name) for field in fields(self)}
+
+    def keys(self) -> KeysView[str]:
+        return self._dict_repr().keys()
+
+    def items(self) -> ItemsView[str, Any]:
+        return self._dict_repr().items()
+
+    def values(self) -> ValuesView[str]:
+        return self._dict_repr().values()
+
+
+@dataclass(frozen=True)
+class TupleLikeAccess:
+    """Useful base class for replacing tuples with dataclass instances and keeping
+    backward compatability regarding read access to the data structure."""
+
+    def __getitem__(self, index: int | slice) -> Any:
+        field_values = [getattr(self, field.name) for field in fields(self)]
+        return field_values[index]
+
+    def __len__(self) -> int:
+        return len(fields(self))
+
+    def __iter__(self) -> Iterator[str]:
+        for field in fields(self):
+            yield getattr(self, field.name)
+
+
+class OptimizationType(str, Enum):
```

Comment:
I have created this same Enum but called it `Direction` because `OptimizationType` could also be multistart vs. single and other things.

Reply:
I would keep it like it is here, and would like to ask you to resolve the conflict in the upcoming PR:)